ast/annotations.go

```diff
@@ -542,3 +565,107 @@ func mergeAnnotations(a *Annotations, b *Annotations) *Annotations {
 
 	return result
 }
+
+func getPackageAnnotations(as *annotationSet, pkg *Package) *Annotations {
+	subPkgAnnot := as.getSubpackagesScope(pkg.Path)
+
+	result := make([]*Annotations, 0, len(subPkgAnnot)+1)
+
+	result = append(result, subPkgAnnot...)
+
+	if x := as.getPackageScope(pkg); x != nil {
+		result = append(result, x)
+	}
+
+	return mergeAnnotationsList(result)
+}
+
+func getRuleAnnotations(as *annotationSet, rule *Rule) *Annotations {
+	ruleAnnot := as.getRuleScope(rule)
+
+	result := make([]*Annotations, 0, len(ruleAnnot)+2)
+
+	if a := getPackageAnnotations(as, rule.Module.Package); a != nil {
+		result = append(result, a)
+	}
+
+	if a := as.getDocumentScope(rule.Path()); a != nil {
+		result = append(result, a)
+	}
+
+	result = append(result, ruleAnnot...)
+
+	return mergeAnnotationsList(result)
+}
+
+type AnnotationsRef struct {
+	Location    *Location
+	Path        Ref
+	Annotations *Annotations
+}
+
+func (ar *AnnotationsRef) MarshalJSON() ([]byte, error) {
+	tmp := map[string]interface{}{
+		"location": ar.Location.String(),
+		"path":     ar.Path.String(),
+	}
+
+	if ar.Annotations != nil {
+		tmp["annotations"] = ar.Annotations
+	}
+
+	return json.Marshal(tmp)
+}
+
+// GetAnnotations returns a list of annotations for every Package and Rule in the passed modules.
+func GetAnnotations(modules []*Module, paths ...Ref) ([]*AnnotationsRef, Errors) {
+	var refs []*AnnotationsRef
+
+	as, err := buildAnnotationSet(modules)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, m := range modules {
+		if len(paths) == 0 || contains(paths, m.Package.Path) {
+			a := getPackageAnnotations(as, m.Package)
+			refs = append(refs, &AnnotationsRef{
+				Location:    m.Package.Location,
+				Path:        m.Package.Path,
+				Annotations: a,
+			})
+		}
+
+		for _, r := range m.Rules {
+			p := r.Path()
+			if len(paths) == 0 || contains(paths, p) {
+				a := getRuleAnnotations(as, r)
+				refs = append(refs, &AnnotationsRef{
+					Location:    r.Location,
+					Path:        p,
+					Annotations: a,
+				})
+			}
+		}
+	}
+
+	return refs, nil
+}
+
+func contains(rs []Ref, ref Ref) bool {
+	for _, r := range rs {
+		if r.Equal(ref) {
+			return true
+		}
+	}
+	return false
+}
+
+func FindPackageAnnotations(as []*Annotations) *Annotations {
```

Comment:
💭 I was confused for a short moment by the _plural_ -- since we only ever have one package annotation in the `as` slice and the code returns early instead of collection the matches. But the thing we return is _multiple annotations_ in the one struct, so that's accurate.